ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/UserManagementStorageBackendImpl.scala

```diff
@@ -121,27 +139,44 @@ object UserManagementStorageBackendImpl extends UserManagementStorageBackend {
       }
   }
 
-  override def getUsersOrderedById(fromExcl: Option[UserId], maxResults: Int)(
+  override def getUsersOrderedById(
+      fromExcl: Option[UserId],
+      maxResults: Int,
+      filter: ListUsersFilter,
+  )(
       connection: Connection
   ): Vector[UserManagementStorageBackend.DbUserWithId] = {
     import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
-    val whereClause = fromExcl match {
-      case None => cSQL""
-      case Some(id: String) => cSQL"WHERE user_id > ${id}"
+    val whereClause = (fromExcl, filter) match {
+      case (None, ListUsersFilter.Wildcard) => cSQL""
+      case (Some(id: String), ListUsersFilter.ByIdentityProviderId(identityProviderId)) =>
+        cSQL"WHERE user_id > ${id} AND identity_provider_id=${identityProviderId.value: String}"
```

Comment:
I do not follow... `identityProviderId.value` will always be some non-empty String..so it will by always some concrete value, or do I miss something?

Reply:
Ah, so `identity_provider_id` in the db can be NULL but `identityProviderId.value` coming from JVM will never be.
I'd suggest adding a comment clarifying that we don't have to guard against nulls because `identityProviderId.value` is guaranteed to be a non-empty string

Reply:
How would one query all the users assigned to the default identity provider and only them?

Reply:
so as agreed offline, currently we dont provide a way to filter users by default IDP. Default IDP user who is an admin will be able to see all users, regardless of their IDP. Users who are IDP Admins - will see only users within their IDP.